File: knowledge_base/meals.pl
%%  meals.pl  —  Meal Database (15 cafeteria meals)
%%  ─────────────────────────────────────────────────
%%  Pure structured facts — NO logic lives here.
%%
%%  meal(+ID, +Name, +Category, +NutritionGoal, +GymSuitable, +MealType).
%%
%%  Category     :  veg | non_veg
%%  Nutrition    :  high_protein | low_carb | low_fat | balanced
%%  GymSuitable  :  yes | moderate | no
%%  MealType     :  breakfast | lunch | dinner | snack

meal(m1,  'Lentil Salad with Boiled Eggs and Nuts',
     veg,      high_protein, yes,      lunch).

meal(m2,  'Grilled Chicken with Brown Rice',
     non_veg,  high_protein, yes,      lunch).

meal(m3,  'Vegetable Stir Fry with Tofu',
     veg,      high_protein, yes,      dinner).

meal(m4,  'Fried Rice with Chicken',
     non_veg,  balanced,     no,       lunch).

meal(m5,  'Oatmeal with Fruits and Nuts',
     veg,      balanced,     yes,      breakfast).

meal(m6,  'Baked Fish with Steamed Vegetables',
     non_veg,  low_carb,     yes,      dinner).

meal(m7,  'Beans with Plantain',
     veg,      balanced,     moderate, lunch).

meal(m8,  'Fruit Salad',
     veg,      low_carb,     moderate, snack).

meal(m9,  'Scrambled Eggs with Avocado Toast',
     veg,      high_protein, yes,      breakfast).

meal(m10, 'Turkey Wrap with Vegetables',
     non_veg,  balanced,     yes,      lunch).

meal(m11, 'Greek Yogurt with Granola and Honey',
     veg,      balanced,     moderate, breakfast).

meal(m12, 'Beef Stir Fry with Mixed Vegetables',
     non_veg,  high_protein, yes,      dinner).

meal(m13, 'Chickpea and Quinoa Bowl',
     veg,      high_protein, yes,      lunch).

meal(m14, 'Protein Smoothie with Banana and Peanut Butter',
     veg,      high_protein, yes,      snack).

meal(m15, 'Grilled Tilapia with Salad',
     non_veg,  low_carb,     yes,      dinner).

File: interface/cafeteria-menu-advisor/src/data/meals.js
export const MEAL_DB = {
  m1:  { name: "Lentil Salad with Boiled Eggs & Nuts",        cat: "veg",     cal: "Medium", type: "lunch",     badge: "High Protein" },
  m2:  { name: "Grilled Chicken with Brown Rice",             cat: "non_veg", cal: "High",   type: "lunch",     badge: "High Protein" },
  m3:  { name: "Vegetable Stir Fry with Tofu",                cat: "veg",     cal: "Medium", type: "dinner",    badge: "High Protein" },
  m4:  { name: "Fried Rice with Chicken",                     cat: "non_veg", cal: "High",   type: "lunch",     badge: "Balanced"     },
  m5:  { name: "Oatmeal with Fruits & Nuts",                  cat: "veg",     cal: "Medium", type: "breakfast", badge: "Balanced"     },
  m6:  { name: "Baked Fish with Steamed Vegetables",          cat: "non_veg", cal: "Medium", type: "dinner",    badge: "Low Carb"     },
  m7:  { name: "Beans with Plantain",                         cat: "veg",     cal: "High",   type: "lunch",     badge: "Balanced"     },
  m8:  { name: "Fruit Salad",                                 cat: "veg",     cal: "Low",    type: "snack",     badge: "Low Carb"     },
  m9:  { name: "Scrambled Eggs with Avocado Toast",           cat: "veg",     cal: "Medium", type: "breakfast", badge: "High Protein" },
  m10: { name: "Turkey Wrap with Vegetables",                 cat: "non_veg", cal: "Medium", type: "lunch",     badge: "Balanced"     },
  m11: { name: "Greek Yogurt with Granola & Honey",           cat: "veg",     cal: "Medium", type: "breakfast", badge: "Balanced"     },
  m12: { name: "Beef Stir Fry with Mixed Vegetables",         cat: "non_veg", cal: "High",   type: "dinner",    badge: "High Protein" },
  m13: { name: "Chickpea and Quinoa Bowl",                    cat: "veg",     cal: "Medium", type: "lunch",     badge: "High Protein" },
  m14: { name: "Protein Smoothie (Banana & Peanut Butter)",   cat: "veg",     cal: "Medium", type: "snack",     badge: "High Protein" },
  m15: { name: "Grilled Tilapia with Salad",                  cat: "non_veg", cal: "Low",    type: "dinner",    badge: "Low Carb"     },
};

export const ALLERGY_OPTIONS = [
  { value: "nuts",   label: "Nuts",   icon: "🥜" },
  { value: "dairy",  label: "Dairy",  icon: "🥛" },
  { value: "eggs",   label: "Eggs",   icon: "🥚" },
  { value: "fish",   label: "Fish",   icon: "🐟" },
  { value: "gluten", label: "Gluten", icon: "🌾" },
  { value: "soy",    label: "Soy",    icon: "🫘" },
];

export const STEPS = ["category", "goal", "health", "meal_type", "convenience"];

export const STEP_CONFIG = {
  category: {
    label: "Who are you?",
    sub: "Tell us about your dietary preference",
    options: [
      { value: "veg",     label: "Vegetarian",      icon: "🥗", desc: "Plant-based meals only"    },
      { value: "non_veg", label: "Non-Vegetarian",  icon: "🍗", desc: "All meals available"       },
      { value: "gym",     label: "Gym Student",     icon: "💪", desc: "High-performance fuel"     },
      { value: "none",    label: "No Restriction",  icon: "🍽️", desc: "Everything goes"           },
    ],
  },
  goal: {
    label: "Nutritional Goal",
    sub: "What are you optimizing for?",
    options: [
      { value: "high_protein", label: "High Protein",  icon: "🥩", desc: "Build & repair muscle"   },
      { value: "low_carb",     label: "Low Carb",      icon: "🥦", desc: "Reduce carbohydrates"    },
      { value: "low_fat",      label: "Low Fat",       icon: "🫐", desc: "Keep fats in check"      },
      { value: "balanced",     label: "Balanced Diet", icon: "⚖️", desc: "Well-rounded nutrition"  },
    ],
  },
  health: {
    label: "Health Objective",
    sub: "Any specific health goal?",
    options: [
      { value: "wl",   label: "Weight Loss",      icon: "📉", desc: "Lean, calorie-smart meals" },
      { value: "wg",   label: "Weight Gain",      icon: "📈", desc: "Calorie-dense options"     },
      { value: "none", label: "No Specific Goal", icon: "😌", desc: "Just eat well"             },
    ],
  },
  meal_type: {
    label: "Meal Time",
    sub: "Which meal are you planning?",
    options: [
      { value: "breakfast", label: "Breakfast", icon: "🌅", desc: "Start your day right"    },
      { value: "lunch",     label: "Lunch",     icon: "☀️", desc: "Midday energy boost"     },
      { value: "dinner",    label: "Dinner",    icon: "🌙", desc: "Evening nourishment"     },
      { value: "snack",     label: "Snack",     icon: "🍌", desc: "Quick bite between meals"},
    ],
  },
  convenience: {
    label: "Convenience",
    sub: "How much time do you have?",
    options: [
      { value: "quick", label: "Quick Meal",     icon: "⚡", desc: "Ready in minutes"       },
      { value: "none",  label: "No Preference",  icon: "🕐", desc: "Time is not a concern"  },
    ],
  },
};

export const MEAL_EMOJI = {
  breakfast: "🍳", lunch: "🥘", dinner: "🍲", snack: "🍎",
};

export const CAL_COLOR = {
  Low:    "text-emerald-600 bg-emerald-50",
  Medium: "text-amber-600 bg-amber-50",
  High:   "text-rose-600 bg-rose-50",
};

export const BADGE_COLOR = {
  "High Protein": "bg-blue-100 text-blue-700",
  "Low Carb":     "bg-green-100 text-green-700",
  "Balanced":     "bg-purple-100 text-purple-700",
  "Low Fat":      "bg-orange-100 text-orange-700",
};
